### PR TITLE
replacement callback for current early stopping

### PR DESCRIPTION
**Pull request by theo1701**
this new callback will replace the old early stopping callback. It will lock the parallel models that are not improving by restoring their best weight and setting them to untrainable, effetely freeze them in their best result.

### CODE REVIEW

**Comment by theo1701** [comment]
The basic functionality is finished and runs. It requires more testing for sure. I'm also considering integrating the checkpoint into this callback since it is also currently judging best monitor value by the sum

**Review by zctao — COMMENTED** [review]
Could you rebase this branch to the current master and resolve the conflicts? This is good to go in afterwards.

File: python/modelUtils.py [hunk 1]
```diff
@@ -3,14 +3,16 @@
 """
 # FIXME: Fix support for other types of networks, currently only default dense network works
 
+import csv
```
[the remark this answers is not in the document]
Comment (theo1701) [reply python/modelUtils.py]: Nope. Removed.

**Comment by zctao** [comment]
It now crashes with:
```ValueError: Missing data for input "model_0_input". You passed a data dictionary with keys ['input_0']. Expected the following keys: ['model_0_input']```
Looks like there is some naming incompatibility between this one and #48.

**Comment by theo1701** [comment]
The input dictionary was using a different naming convention at the time. Fixed now. Everything now uses the naming convention from `layer_namer`.

**Pull request merged**